tests/bcl-test/BCLTests/templates/common/TestRunner.NUnit/TestMethodFilter.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+
+using NUnit.Framework.Api;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Filters;
+
+namespace Xamarin.iOS.UnitTests.NUnit
+{
+	public class TestMethodFilter : TestFilter
+	{
+		List <string> methods;
+
+		public TestMethodFilter (string method)
+		{
+			AddMethod (method);
+		}
+
+		public TestMethodFilter (IEnumerable<string> methods)
+		{
+			if (methods == null)
+				throw new ArgumentNullException (nameof (methods));
+
+			foreach (string m in methods) {
+				AddMethod (m);
+			}
+		}
+
+		public void AddMethod (string method)
+		{
+			if (string.IsNullOrEmpty (method))
+				throw new ArgumentException ("must not be null or empty", nameof (method));
+
+			if (methods == null)
+				methods = new List <string> ();
+			if (methods.Contains (method))
+				return;
```

Comment:
checking for dupe could be doe while loading the entries (from the file)

Reply:
Yes, and no. In the sense, we want to share later this with android, and we do not know if they will use *.ignore files as we do or not. I'll ping them about it. If they use *ignore files, well, we can do it over there.

Reply:
good point

Reply:
As your `.ignore` files are `@(EmbeddedResource)` entries, Android *can* use them.  (If they were `@(Content)`, that wouldn't work, as Android doesn't support `@(Content)`.)

Android already uses a *similar* mechanism, looking specifically for the [`nunit-excluded-tests.txt`](https://github.com/xamarin/xamarin-android/blob/f5132708f0652a0f29eef6e832ae7f21cebd71dc/tests/BCL-Tests/Xamarin.Android.Bcl-Tests/NUnitInstrumentation.cs#L63) or [`xunit-excluded-tests.txt`](https://github.com/xamarin/xamarin-android/blob/master/tests/BCL-Tests/Xamarin.Android.Bcl-Tests/XUnitInstrumentation.cs#L52) resources.

A question though: do we want/need *separate* lists for NUnit vs. xUnit?  Android currently has separate lists, but I don't know why, and @grendello is currently on vacation.

If we *don't* need separate lists, then `*.ignore` resources should be fine.

If we *do* want separate lists, we may want a different naming convention to separate them from each other.

A different issue with `*.ignore` isn't the mechanics in this PR, but the mechanics of git, and mixed macOS/Windows use.  By default, `*.ignore` won't be treated as a "text" file (unless you update `.gitattributes`), while `.txt` *will* be treated as text.

As such, perhaps instead of `*.ignore` it might be better to use `*.ignore.txt`?

Reply:
@jonpryor we use *.ignore because we have the same ignoring pattern in other tests from iOS, but it we want to share this, I personally have no issue in changing the naming to *.ignore.txt (as long as @spouliot is ok with that).

We do not make a difference in the ignore files names between nunit and xunit because, when we generate the test projects, we know in advance if we are dealing with a NUnit test project or a xUnit one, since we know that:

* "*_test.dll" -> NUnit
* "*_xunit-test.dll" -> xUnit

So we have a list per testing project (to make it easier to maintain). I suppose that the difference in your side resides in the fact that you do not AFAIK have a filter that works on methods for the NUnit runner (something that I am adding in this PR). 

About resource files, I can add a method to load from @(Content) and other for resource files, each platform can decide how the files are added. We could use content to speed our rebuild times and leave the resources file approach there for the android team.

Reply:
Let me know your opinion and I'll do push regarding the use or resource. All other comments have been addressed :)

Reply:
The extension name is not important, it just needs to be consistent (at least across unit tests).